app/models/miq_event.rb

```diff
@@ -34,13 +30,42 @@ def self.description
     _("Policy Events")
   end
 
+  def self.group_name(group)
+    return if group.nil?
+
+    group_names_and_levels.dig(:group_names, group.to_sym) || DEFAULT_GROUP_NAME_STR
+  end
+
+  def self.group_and_level(event_type)
+    by_literal = partition_group_and_level_by_event_type
+    by_literal[event_type.to_s] || [DEFAULT_GROUP_NAME, DEFAULT_GROUP_LEVEL]
+  end
+
+  private_class_method def self.partition_group_and_level_by_event_type
```

Comment:
Is this the same as the base class method?  If so can we just reuse it?  If not can we do it as a mixin?

Reply:
No, it's close but not the same... See the `MiqEventDefinition` part below.